python/lsst/ts/envsensors/sensor/wind_sensor.py

```diff
@@ -0,0 +1,136 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "WindSensor",
+    "DEFAULT_DIRECTION_VAL",
+    "DEFAULT_SPEED_VAL",
+    "DEFAULT_STATUS",
+    "END_CHARACTER",
+    "START_CHARACTER",
+    "UNIT_IDENTIFIER",
+    "WINDSPEED_UNIT",
+]
+
+import asyncio
+import logging
+import math
+from typing import List, Optional, Union
+
+from ..constants import DISCONNECTED_VALUE
+from ..response_code import ResponseCode
+from .base_sensor import BaseSensor, DELIMITER, TERMINATOR
+
+"""ASCII start character."""
+START_CHARACTER: str = "\x02"
+
+"""Unit Identifier."""
+UNIT_IDENTIFIER: str = "Q"
+
+"""Windspeed unit."""
+WINDSPEED_UNIT: str = "M"
+
+"""Default status."""
+DEFAULT_STATUS: str = "00"
+
+"""ASCII end charactor."""
+END_CHARACTER = "\x03"
+
+"""Default value for the wind direction."""
+DEFAULT_DIRECTION_VAL: float = 999
+
+"""Default value for the wind speed."""
+DEFAULT_SPEED_VAL: float = 9999.9990
+
+
+class WindSensor(BaseSensor):
+    """Wind Sensor.
+
+    Perform protocol conversion for Gill Windsonic Ultrasonic Anemometer
+    instrument. The instrument is assumed to use its default message format -
+    "Gill - Polar, Continuous" as documented in Gill Windsonic Doc No 1405 PS
+    0019 Issue 28.
+    Serial data is output by the anemometer once per second with the following
+    format:
+
+        <STX>Q,ddd,sss.ss,M,00,<ETX>checksum<CR><LF>'
+
+    where:
+
+        <STX>       ASCII start character.
+        'Q'         Unit Identifier ('Q' is default value).
+        ddd         Wind direction. Three character, leading zero's integer.
+                    000-359 degrees. Wind direction value is empty ('') when
+                    wind speed is below 0.05 m/s.
+        sss.ss      Wind speed. Six character, floating point, leading zero's.
+                    0 to 60 m/s.
+        'M'         Unit of speed measurement ('M' is m/s default)
+        '00'        Status.
+        <ETX>       ASCII end charactor.
+        checksum    Exclusive OR of all bytes in the string between <STX> and
+                    <ETX> characters.
+        <CR><LF>    2-character terminator ('\r\n').
+
+    Parameters
+    ----------
+    name: `str`
+        The name of the sensor.
+    channels: `int`
+        The number of temperature channels.
+    log: `logger`' optional
+        The logger for which to create a child logger, or None in which case a
+        new logger gets requested.
+    """
+
+    def __init__(
+        self,
+        channels: int,
+        log: logging.Logger,
+    ) -> None:
+        super().__init__(channels=channels, log=log)
+
+    async def extract_telemetry(self, line: str) -> List[float]:
+        """Extract the wind telemetry from a line of Sensor data."""
+        resp = line.strip(TERMINATOR)
+        data = resp.split(DELIMITER)
+        speed = DEFAULT_SPEED_VAL
+        direction = DEFAULT_DIRECTION_VAL
+        if len(data) == 6:
```

Comment:
I suggest you use a regex to parse this string, instead of all the different constants. You can have named fields for speed, direction, and status and parse the whole thing with a single call -- no need to strip the terminator, split the string, or anything. Then:
* If the string doesn't parse, log "cannot be parsed"
* If the status is not 00 then log "status not OK"
* Otherwise you know the data is valid; parse it and return it.

That makes it much easier to log problems with invalid data.
It also eliminates that long list of constants -- you can just compile a regex in the constructor and use that instead.

I am generally a bit leery of using regex, but this seems an excellent use case.

Reply:
Done.